### PR TITLE
Add four unit tests for aws/utils

### PR DESCRIPTION
**Pull request by slycyberguy**
Unit tests for:

airflow/provider/amazon/aws/utils/sagemaker.py
airflow/provider/amazon/aws/utils/rds.py
airflow/provider/amazon/aws/utils/sqs.py
airflow/provider/amazon/aws/utils/tags.py

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/providers/amazon/aws/utils/test_rds.py
Comment (hussein-awala) [inline tests/providers/amazon/aws/utils/test_rds.py:1]: This test is useless

File: tests/providers/amazon/aws/utils/test_sagemaker.py
Comment (hussein-awala) [inline tests/providers/amazon/aws/utils/test_sagemaker.py:1]: same here

Comment (potiuk) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: I had the same impression that most of the tests produced / generated here are useless. I think it's not a good idea to clutter and make more of those tests that are really not testing anything - that's why in the other PR I've asked whether the tests were generated.

I think all those PRs where the tests were generated to start with and then replicated should not make it into our codebase.

Comment (potiuk) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: Potential Value added vs. time needed to review/classify those tests do not seem to be worth it.

Comment (slycyberguy) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: I removed those tests. I figured they were necessary because those services were listed in issue #35442

Comment (o-nikolas) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: I didn't review the previous tests that have been removed from this PR, but the tests remaining seem to add coverage. Hard to tell for sure if they were generated (there are some weird names like `response_dict_dict` that seem unnatural).

**Review by o-nikolas — COMMENTED** [review]
I didn't review the previous tests that have been removed from this PR, but the tests remaining seem to add coverage. Hard to tell for sure if they were generated (there are some weird names like response_dict_dict that seem unnatural).

File: tests/providers/amazon/aws/utils/test_sagemaker.py
Comment (slycyberguy) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: It's not generated. I tried to come up with a more descriptive name for the message samples. One was a list of dict and the other was a list of dict of dict. That's why I chose those names. Anyway, I just renamed the fixture labels to sample_message_1 and sample_message_2. Hope that works. Let me know your suggestions.

Comment (o-nikolas) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: I think maybe `response_dict_json` would be more accurate since we're parsing it as json. Or maybe `response_nested_dict` if we really want to keep it dict.

But any who, naming aside. Thanks for adding the tests :slightly_smiling_face:  and sorry you got some suspicious push back from us!

Comment (slycyberguy) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: No problem. Thanks for the suggestions. I renamed them similar to your examples.

**Review by o-nikolas — APPROVED** [review]

File: tests/providers/amazon/aws/utils/test_sagemaker.py
Comment (o-nikolas) [reply tests/providers/amazon/aws/utils/test_sagemaker.py:1]: Sounds good! I approved. We'll need have all conversations resolved before merging. I'll let @potiuk and @hussein-awala reply back to this thread to see if they have any concerns.

**Review by o-nikolas — COMMENTED** [review]
You will need to remove these lines from the project structure tests:

https://github.com/apache/airflow/blob/main/tests/always/test_project_structure.py#L85-L86

**Comment by slycyberguy** [comment]
> You will need to remove these lines from the project structure tests:
> 
> https://github.com/apache/airflow/blob/main/tests/always/test_project_structure.py#L85-L86

I took care of it

**Comment by slycyberguy** [comment]
> You will need to remove these lines from the project structure tests:
> 
> https://github.com/apache/airflow/blob/main/tests/always/test_project_structure.py#L85-L86

Do you know if the others will be reviewing this PR any time soon? Also [PR 38819](https://github.com/apache/airflow/pull/38819) has been approved by one reviewer and since you're named as one of the reviewers, I was hoping you can review it when you get a chance. Thanks.

**Comment by o-nikolas** [comment]
> > You will need to remove these lines from the project structure tests:
> > https://github.com/apache/airflow/blob/main/tests/always/test_project_structure.py#L85-L86
> 
> Do you know if the others will be reviewing this PR any time soon? Also [PR 38819](https://github.com/apache/airflow/pull/38819) has been approved by one reviewer and since you're named as one of the reviewers, I was hoping you can review it when you get a chance. Thanks.

It's not really about the number of approvers, it was more that @potiuk and @hussein-awala had some dissenting opinions on the PR, so I wanted to give them the opportunity to review the current version after your changes and see if they find it more acceptable now. However, they do seem to be busy! So if no one checks in by end of today or tomorrow I can merge this. We can always make changes later :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Now those tests looked  more useful

**Pull request merged**
